app/ide-desktop/client/src/projectManagement.ts

```diff
@@ -460,3 +464,40 @@ export function bumpMetadata(
   })).id
   return { id, name, parentDirectory }
 }
+
+/** Download project templates GitHub repo into the Samples directory if one not exists. */
+export async function downloadSamples(): Promise<void> {
+  logger.log('Downloading samples.')
+
+  const samplesDirectory = pathModule.join(getProjectsDirectory(), SAMPLES_DIRECTORY_NAME)
+
+  return new Promise((resolve, reject) => {
+    fs.access(samplesDirectory, fs.constants.F_OK, (err) => {
+      if (err == null) {
+        return resolve()
+      }
+      fs.mkdir(samplesDirectory, { recursive: true }, (err) => {
+        if (err != null) {
+          logger.error(err)
+          return reject(err)
+        }
+        https.get(SAMPLES_URL, (redirectResponse) => {
```

Comment:
FWIW, you can now use `fetch` api in nodejs

Reply:
For now, `fetch` does not allow you to pipe the response easily. You have to manually convert the body to `ReadableStream` https://developer.mozilla.org/en-US/docs/Web/API/Streams_API/Using_readable_streams#reading_the_stream